src/plugins/integrations/MySQL2Integration.ts

```diff
@@ -40,68 +42,73 @@ class MySQL2Integration implements Integration {
-            [DBTags.DB_STATEMENT]: statement,
-            [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ? SQLQueryOperationTypes[statementType] : 'READ',
+            [SpanTags.SPAN_TYPE]: SpanTypes.RDB,
+            [DBTags.DB_INSTANCE]: this.config.database,
+            [DBTags.DB_USER]: this.config.user,
+            [DBTags.DB_HOST]: this.config.host,
+            [DBTags.DB_PORT]: this.config.port,
+            [DBTags.DB_TYPE]: DBTypes.MYSQL,
+            [SpanTags.TOPOLOGY_VERTEX]: true,
+            [SpanTags.TRIGGER_DOMAIN_NAME]: LAMBDA_APPLICATION_DOMAIN_NAME,
+            [SpanTags.TRIGGER_CLASS_NAME]: LAMBDA_APPLICATION_CLASS_NAME,
+            [SpanTags.TRIGGER_OPERATION_NAMES]: [tracer.functionName],
           });
-        }
 
-        const originalCallback = sequence.onResult;
+          const sequence = query.call(this, sql, values, cb);
 
-        const wrappedCallback = (err: any, res: any) => {
-          if (err) {
-            const parseError = Utils.parseError(err);
-            span.setTag('error', true);
-            span.setTag('error.kind', parseError.errorType);
-            span.setTag('error.message', parseError.errorMessage);
-            span.setTag('error.stack', parseError.stack);
-            span.setTag('error.code', parseError.code);
+          const statement = sequence.sql;
+
+          if (statement) {
+            const statementType = statement.split(' ')[0].toUpperCase();
+            span.addTags({
+              [DBTags.DB_STATEMENT_TYPE]: statementType,
+              [DBTags.DB_STATEMENT]: statement,
+              [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ? SQLQueryOperationTypes[statementType] : 'READ',
+            });
           }
 
-          span.close();
+          const originalCallback = sequence.onResult;
 
-          originalCallback(err, res);
-        };
+          const wrappedCallback = (err: any, res: any) => {
+            if (err) {
+              const parseError = Utils.parseError(err);
+              span.setTag('error', true);
+              span.setTag('error.kind', parseError.errorType);
+              span.setTag('error.message', parseError.errorMessage);
+              span.setTag('error.stack', parseError.stack);
+              span.setTag('error.code', parseError.code);
+            }
 
-        if (sequence.onResult) {
-          sequence.onResult = wrappedCallback;
-        } else {
-          sequence.on('end', () => {
             span.close();
-          });
-        }
 
-        return sequence;
+            originalCallback(err, res);
+          };
+
+          if (sequence.onResult) {
+            sequence.onResult = wrappedCallback;
+          } else {
+            sequence.on('end', () => {
+              span.close();
+            });
+          }
+
+          return sequence;
+        } catch (error) {
+          ThundraLogger.getInstance().error(error);
+          query.call(this, sql, values, cb);
```

Comment:
Can we be sure that in case of error `query.call` is not called twice

Reply:
In case of error in the wrapped function error callback is called. I verified, it is not called twice.